src/wasi.ts

```diff
@@ -813,9 +813,59 @@ export default class WASI {
           return wasi.ERRNO_BADF;
         }
       },
-      // eslint-disable-next-line @typescript-eslint/no-unused-vars
-      poll_oneoff(in_, out, nsubscriptions) {
-        throw "async io not supported";
+      async poll_oneoff(
+        in_ptr: number,
+        out_ptr: number,
+        nsubscriptions: number,
+      ): Promise<number> {
+        if (nsubscriptions === 0) {
+          return wasi.ERRNO_INVAL;
+        }
+        // TODO: For now, we only support a single subscription just to be enough for wasi-libc's
+        // clock_nanosleep.
+        if (nsubscriptions > 1) {
+          return wasi.ERRNO_NOTSUP;
```

Comment:
Maybe add a `debug.log()` for each `ERRNO_NOTSUP` error explaining what the error is?